Ginger/GingerCoreNET/Drivers/CoreDrivers/Mobile/Appium/GenericAppiumDriver.cs

```diff
@@ -252,23 +252,36 @@ public bool ConnectToAppium()
                         break;
                 }
 
-                mSeleniumDriver = new SeleniumDriver(Driver); //used for running regular Selenium actions
-                mSeleniumDriver.StopProcess = this.StopProcess;
-                mSeleniumDriver.BusinessFlow = this.BusinessFlow;
-
-                if (AppType == eAppType.Web && mDefaultURL != null)
+                //If Driver.SessionId is null, it means that the Mobile Agent already in use.
+                
+                if (!(Driver.Capabilities.HasCapability("message") && Driver.Capabilities.GetCapability("message").ToString() == "Could not find available device"))
```

Comment:
why to add this if and not look for this issue and proper error message only inside the catch block?